galite-demo/galite-vaadin/src/main/kotlin/org/kopi/galite/demo/command/CommandForm.kt

```diff
@@ -102,6 +116,103 @@ class CommandForm : DictionaryForm(title = "Commands", locale = Locale.UK) {
     init {
       blockVisibility(Access.VISIT, Mode.QUERY)
     }
+
+    val PostqryTrigger = trigger(POSTQRY) {
+      purchases.idClt[0] = idClt.value
+      purchases.load()
+      for (rec in 0 until purchases.block.bufferSize) {
+        if (purchases.block.isRecordFilled(rec)) {
+          purchases.colorRecord(rec)
+        }
+      }
+    }
+
+    /**
+     * Save block
+     */
+    fun save(b: VBlock) {
+      tb1.block.validate()
+
+      if (!purchases.isFilled()) {
+        purchases.currentRecord = 0
+        throw VExecFailedException("Purchase block is empty.")
+      }
+
+      transaction {
+        tb1.block.save()
+        purchases.block.save()
+      }
+
+      b.form.reset()
+    }
+  }
+
+  inner class BlockPurchase : Block("Purchase", 10, 10) {
+    val u = table(Purchase)
+
+    val numPurchase = hidden(domain = INT(20)) {
+      label = "Number"
+      help = "The purchase number"
+      columns(u.id)
+    }
+    val idClt = mustFill(domain = INT(25), position = at(1, 1)) {
+      label = "Client ID"
+      help = "The client ID"
+      columns(u.idClt)
+    }
+    val idProduct = mustFill(domain = INT(20), position = at(2, 1)) {
+      label = "Product ID"
+      help = "The Product ID"
+      columns(u.idPdt)
+    }
+    val quantity = mustFill(domain = INT(7), position = at(4, 1)) {
+      label = "Quantity"
+      help = "quantity of the current purchase"
+      columns(u.quantity)
+    }
+
+    val color = mustFill(domain = COLOR, position = at(4, 1)) {
+      label = "color"
+      help = "color [for test purpose] "
+      columns(u.color)
+    }
+
+    init {
+      blockVisibility(Access.VISIT, Mode.QUERY)
+      options(BlockOption.NODETAIL)
+    }
```

Comment:
I think it's better to add `border = Border.LINE` in the `init` block so that the block name can be shown for a better view of the form